src/Microsoft.ApiDesignGuidelines.Analyzers/Core/NestedTypesShouldNotBeVisible.cs

```diff
@@ -81,6 +81,13 @@ public override void Initialize(AnalysisContext analysisContext)
                                 return;
                             }
 
+                            // Even if the nested type is declared public, don't complain if it's within
```

Comment:
No, that breaks other cases, for example:

```
public class Outer
{
    protected internal class Inner
    {
    }
}
```

This should not give a diagnostic, but if I make the change you suggest, it does give a diagnostic because the inner type is visible outside the assembly (to derived types of `Outer`).